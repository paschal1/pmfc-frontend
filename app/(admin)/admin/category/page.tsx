'use client'
import {
  MdOutlineAddBox,
  MdOutlineEdit,
  MdOutlineRemoveRedEye,
} from 'react-icons/md'
import { useRouter } from 'next/navigation'
import { useEffect, useState } from 'react'
import axios from 'axios'
import Cookies from 'js-cookie'
import Link from 'next/link'
import { RiDeleteBin5Line } from 'react-icons/ri'
import Image from 'next/image'

type CategoryType = {
  id: number
  name: string
  image: string
  thumbnailimage: string
}

const Category = () => {
  const router = useRouter()
  const [category, setCategory] = useState<CategoryType[]>([])
  const [loading, setLoading] = useState(true)
  const [error, setError] = useState('')

  const handleDelete = async (id: number) => {
    try {
      const token = Cookies.get('adminToken')
      if (!token) {
        return
      }

      await axios.delete(`https://api.princem-fc.com/api/categories/${id}`, {
        headers: {
          Authorization: `Bearer ${token}`,
        },
      })

      // Remove deleted category from UI
      setCategory((prev) => prev.filter((cat) => cat.id !== id))
    } catch (error: any) {
      setError(error.response.data.message)
    }
  }

  useEffect(() => {
    const fetchCategory = async () => {
      try {
        const token = Cookies.get('adminToken')

        if (!token) {
          return
        }

        const response = await axios.get(
          'https://api.princem-fc.com/api/categories',
          {
            headers: {
              Authorization: `Bearer ${token}`,
            },
          }
        )
        const categoryData = response.data.data
        setCategory(categoryData)
      } catch (error) {
        return
      }
    }

    fetchCategory()
  }, [])

  return (
    <div className="bg-white min-h-screen w-full flex flex-col pb-[3rem]">
      <div className="xl:ml-[20rem] mt-8 bg-[#F2F2F2] flex flex-col px-4 w-[90%] lg:w-[1014px] rounded-xl mx-auto mb-8 pb-8">
        <div className="flex items-center justify-between mt-4">
          <h1 className="font-semibold sm:text-xl text-lg">All Category</h1>
          <button
            onClick={() => router.push('/admin/add-new-category')}
            className="px-7 py-2 bg-[#fab702] rounded-[5px] text-white text-[13px] font-semibold hover:text-black hover:opacity-75 active:opacity-60 transition-all duration-500 ease-in-out flex items-center justify-center gap-2"
          >
            <MdOutlineAddBox className="h-[20px] w-[20px]" />
            Add New
          </button>
        </div>
        <div className="flex items-center justify-between mt-4">
          <div></div>
          <div>
            <label className="mr-3">Search</label>
            <input
              type="text"
              placeholder=""
              title="search"
              className="border bg-inherit border-black focus:outline-none pl-2 h-[35px] w-[150px] rounded-[4px]"
            />
          </div>
        </div>
        <div className="mt-8 overflow-x-auto relative">
          <table className="min-w-full border-collapse">
            <thead>
              <tr className="text-left bg-gray-200 rounded-[6px] text-[#4A5568]">
                <th className="lg:px-16 px-8 py-2 whitespace-nowrap">
                  Category Name
                </th>
                <th className="lg:px-16 px-8 py-2 whitespace-nowrap">
                  Category Image
                </th>
                <th className="lg:px-16 px-8 py-2 whitespace-nowrap">
                  Category Thumbnail
                </th>
                <th className="lg:px-16 px-8 py-2 whitespace-nowrap">Option</th>
              </tr>
            </thead>
            <tbody>
              {category.map((item) => {
                // const Icon = item.icon
                return (
                  <tr key={item.id} className="even:bg-white odd:bg-[#F2F2F2]">
                    <td className="lg:px-16 px-8 py-3">
                      <h1 className="text-md text-[#4A5568]">{item.name}</h1>
                    </td>
                    <td className="lg:px-16 px-8 py-3">
                      <div className="w-[80px] h-[80px] flex items-center justify-center rounded-xl">
                        <Image
                          width={60}
                          height={60}
                          src={item.image}
                          alt="img"
                          className="h-[60px] w-[60px] object-contain"
                          unoptimized
                        />
                      </div>
                    </td>
                    <td className="lg:px-24 px-16 py-3 ml-4">
                      <div className="w-[80px] h-[80px] flex items-center justify-center rounded-xl">
                        {item.thumbnailimage ? (
                          <Image
                            width={60}
                            height={60}
                            src={item.thumbnailimage}
                            alt="Thumbnail"
                            className="h-[60px] w-[60px] object-contain"
                            unoptimized
                          />
                        ) : (
                          <span className="text-gray-400 text-sm">
                            No Thumbnail
                          </span>
                        )}
                      </div>
                    </td>
                    <td className="lg:px-16 px-8 py-3 flex mt-9 gap-3 relative">
                      <Link href={`/admin/category/${item.id}`}>
                        <MdOutlineRemoveRedEye className="h-[20px] w-[20px] text-purple-400" />
                      </Link>
                      <Link href={`/admin/category/edit/${item.id}`}>
                        <MdOutlineEdit className="h-[20px] w-[20px] text-blue-400" />
                      </Link>
                      <RiDeleteBin5Line
                        onClick={() => handleDelete(item.id)}
                        className="h-[20px] w-[20px] text-red-400 cursor-pointer"
                      />
                    </td>
                  </tr>
                );
              })}
            </tbody>
          </table>
        </div>
      </div>
    </div>
  )
}

export default Category
