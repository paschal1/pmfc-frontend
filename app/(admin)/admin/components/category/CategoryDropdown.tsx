import { AnimatePresence, motion } from 'framer-motion'
import React from 'react'
import Link from 'next/link'
import { useAppDispatch, useAppSelector } from '../../store/hooks'
import { closeSidebar } from '../../store/mobileSidebarSlice'
import { toggleCategory } from '../../store/sidebarSlice'
import { MdWidgets } from 'react-icons/md'
import { IoIosArrowDown } from 'react-icons/io'

const CategoryDropdown = () => {
  const sections = useAppSelector((state) => state.sidebar)
  const dispatch = useAppDispatch()

  const handleCloseSidebar = () => {
    dispatch(closeSidebar())
  }
  
  const handleCategoryClick = () => {
    dispatch(toggleCategory())
  }

  return (
    <>
      <div
        onClick={() => handleCategoryClick()}
        className="flex items-center justify-between w-[230px]"
      >
        <div className="flex flex-row items-center justify-between gap-8">
          <MdWidgets className="h-[20px] w-[20px]" />
          <h1>Category</h1>
        </div>
        <div className="">
          <motion.div
            animate={{ rotate: sections.category ? 180 : 0 }}
            initial={{ rotate: 0 }}
            transition={{ duration: 0.3 }}
          >
            <IoIosArrowDown className="cursor-pointer" />
          </motion.div>
        </div>
      </div>
      <AnimatePresence>
        {sections.category && (
          <motion.div
            initial={{ height: 0, opacity: 0 }}
            animate={{ height: 'auto', opacity: 1 }}
            exit={{ height: 0, opacity: 0 }}
            transition={{ duration: 0.3 }}
            className="overflow-hidden flex flex-col gap-4 mt-2 ml-14"
          >
            <Link href={'/admin/category'} onClick={() => handleCloseSidebar()}>
              Category List
            </Link>
            <Link
              href={'/admin/add-new-category'}
              onClick={() => handleCloseSidebar()}
            >
              Add New Category
            </Link>
          </motion.div>
        )}
      </AnimatePresence>
    </>
  )
}

export default CategoryDropdown
