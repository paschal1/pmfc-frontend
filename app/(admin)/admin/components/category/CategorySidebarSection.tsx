import React from 'react'
import { AnimatePresence, motion } from 'framer-motion'
import { MdWidgets } from 'react-icons/md'
import { IoIosArrowDown } from 'react-icons/io'
import Link from 'next/link'
import { useAppDispatch, useAppSelector } from '../../store/hooks'
import { toggleCategory } from '../../store/sidebarSlice'

const CategorySidebarSection = () => {
  const sections = useAppSelector((state) => state.sidebar)
  const dispatch = useAppDispatch()

  const handleCategoryClick = () => {
    dispatch(toggleCategory())
  }
  return (
    <>
      <div
        onClick={() => handleCategoryClick()}
        className="flex items-center justify-between w-[230px] cursor-pointer"
      >
        <div className="flex flex-row items-center justify-between gap-8">
          <MdWidgets className="h-[20px] w-[20px]" />
          <h1>Category</h1>
        </div>
        <div className="">
          <motion.div
            animate={{ rotate: sections.category ? 180 : 0 }}
            initial={{ rotate: 0 }}
            transition={{ duration: 0.3 }}
          >
            <IoIosArrowDown className="cursor-pointer" />
          </motion.div>
        </div>
      </div>
      <AnimatePresence>
        {sections.category && (
          <motion.div
            initial={{ height: 0, opacity: 0 }}
            animate={{ height: 'auto', opacity: 1 }}
            exit={{ height: 0, opacity: 0 }}
            transition={{ duration: 0.3 }}
            className="overflow-hidden flex flex-col gap-4 mt-2 ml-14"
          >
            <Link href={'/admin/category'}>Category List</Link>
            <Link href={'/admin/add-new-category'}>Add New Category</Link>
          </motion.div>
        )}
      </AnimatePresence>
    </>
  )
}

export default CategorySidebarSection
