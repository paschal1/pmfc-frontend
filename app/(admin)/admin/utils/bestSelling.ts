export interface BestSellingItem {
  id: number;
  name: string;
  price: number;
  orders: number;
  stock: number;
  amount: number;
  date: string;
  image: string;
}

export const bestSellingData: BestSellingItem[] = [
  {
    id: 0,
    name: 'Wooden Chair',
    image: '/chair.webp',
    price: 29.00,
    orders: 74,
    stock: 510,
    amount: 1759,
    date: '22-01-2025',
  },
  {
    id: 1,
    name: 'Cupboard',
    image: '/bed.png',
    price: 29.00,
    orders: 73,
    stock: 510,
    amount: 1749,
    date: '22-01-2025',
  },
  {
    id: 2,
    name: 'Sofa',
    image: '/sofa.webp',
    price: 29.00,
    orders: 72,
    stock: 500,
    amount: 1729,
    date: '22-01-2025',
  }
]