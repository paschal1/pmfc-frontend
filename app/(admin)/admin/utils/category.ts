import { IconType } from "react-icons/lib"
import { RiArmchairLine } from 'react-icons/ri'
import { IoBedOutline } from 'react-icons/io5'
import { BiCabinet } from 'react-icons/bi'
import { PiDeskBold } from 'react-icons/pi'
import { RiSofaFill } from 'react-icons/ri'
import { MdOutlineMiscellaneousServices } from 'react-icons/md'

export interface CategoryData {
  id: number
  name: string
  image: string
  icon: IconType
}

export const categories: CategoryData[] = [
  {
    id: 0,
    name: 'Chair',
    image: '/chair.webp',
    icon: RiArmchairLine,
  },
  {
    id: 1,
    name: 'Bed',
    image: '/bed.png',
    icon: IoBedOutline,
  },
  {
    id: 2,
    name: 'Cabinet',
    image: '/cabinet.jpg',
    icon: BiCabinet,
  },
  {
    id: 3,
    name: 'Desk',
    image: '/img5.jpg',
    icon: PiDeskBold,
  },
  {
    id: 4,
    name: 'Sofa',
    image: '/img1.jpg',
    icon: RiSofaFill,
  },
  {
    id: 5,
    name: 'Misc',
    image: '/misc.png',
    icon: MdOutlineMiscellaneousServices,
  },
]