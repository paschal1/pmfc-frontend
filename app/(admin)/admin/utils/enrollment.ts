export interface EnrollmentData {
  id: number
  name: string
  email: string
  contact: number
  studentId: number
  trainingProgramId: number
  enrollmentDate: string
}

export const enrollment: EnrollmentData[] = [
  {
    id: 0,
    name: 'John Doe',
    email: 'johndoe@example.com',
    contact: +1234567890,
    studentId: 1001,
    trainingProgramId: 2001,
    enrollmentDate: '2024-02-01',
  },
  {
    id: 1,
    name: 'Jane Smith',
    email: 'janesmith@example.com',
    contact: +1234567891,
    studentId: 1002,
    trainingProgramId: 2002,
    enrollmentDate: '2024-02-05',
  },
  {
    id: 2,
    name: 'Alice Johnson',
    email: 'alicejohnson@example.com',
    contact: +1234567892,
    studentId: 1003,
    trainingProgramId: 2001,
    enrollmentDate: '2024-02-10',
  },
  {
    id: 3,
    name: 'Bob Williams',
    email: 'bobwilliams@example.com',
    contact: +1234567893,
    studentId: 1004,
    trainingProgramId: 2003,
    enrollmentDate: '2024-02-15',
  },
  {
    id: 4,
    name: 'Charlie Brown',
    email: 'charliebrown@example.com',
    contact: +1234567894,
    studentId: 1005,
    trainingProgramId: 2002,
    enrollmentDate: '2024-02-20',
  },
  {
    id: 5,
    name: 'David Wilson',
    email: 'davidwilson@example.com',
    contact: +1234567895,
    studentId: 1006,
    trainingProgramId: 2003,
    enrollmentDate: '2024-02-25',
  },
  {
    id: 6,
    name: 'Emma Davis',
    email: 'emmadavis@example.com',
    contact: +1234567896,
    studentId: 1007,
    trainingProgramId: 2001,
    enrollmentDate: '2024-03-01',
  },
  {
    id: 7,
    name: 'Frank Miller',
    email: 'frankmiller@example.com',
    contact: +1234567897,
    studentId: 1008,
    trainingProgramId: 2002,
    enrollmentDate: '2024-03-05',
  },
]
