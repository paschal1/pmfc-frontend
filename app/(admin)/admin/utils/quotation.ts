export interface QuotationData {
  id: number
  propertyType: string
  areaSize: string
  measurement: string
  budget: string
  name: string
  email: string
  contact: number
  message: string
}

export const quotation: QuotationData[] = [
  {
    id: 1,
    propertyType: 'Residential',
    areaSize: '1500',
    measurement: 'sqft',
    budget: 'Mid Range',
    name: 'James Carter',
    email: 'james.carter@example.com',
    contact: 1234567890,
    message: 'Looking for a modern and spacious home with wooden interiors.',
  },
  {
    id: 2,
    propertyType: 'Office',
    areaSize: '200',
    measurement: 'm',
    budget: 'High End',
    name: 'Sophia Martins',
    email: 'sophia.martins@example.com',
    contact: 9876543210,
    message: 'Need a premium executive office setup with custom furniture.',
  },
  {
    id: 3,
    propertyType: 'Commercial',
    areaSize: '3000',
    measurement: 'sqft',
    budget: 'Budget Friendly',
    name: 'Michael Johnson',
    email: 'michael.johnson@example.com',
    contact: 1122334455,
    message: 'Looking for affordable furniture for my new commercial space.',
  },
  {
    id: 4,
    propertyType: 'Retail',
    areaSize: '1000',
    measurement: 'ft',
    budget: 'Mid Range',
    name: 'Emily Rodriguez',
    email: 'emily.rodriguez@example.com',
    contact: 5566778899,
    message: 'Need stylish retail store furniture with display shelves.',
  },
  {
    id: 5,
    propertyType: 'Residential',
    areaSize: '250',
    measurement: 'm',
    budget: 'High End',
    name: 'Daniel Smith',
    email: 'daniel.smith@example.com',
    contact: 6677889900,
    message: 'Interested in luxury furniture for my new apartment.',
  },
  {
    id: 6,
    propertyType: 'Other',
    areaSize: '500',
    measurement: 'sqft',
    budget: 'Budget Friendly',
    name: 'Olivia Brown',
    email: 'olivia.brown@example.com',
    contact: 7788990011,
    message: 'Looking for durable and eco-friendly furniture options.',
  },
  {
    id: 7,
    propertyType: 'Commercial',
    areaSize: '120',
    measurement: 'm',
    budget: 'Mid Range',
    name: 'William Anderson',
    email: 'william.anderson@example.com',
    contact: 9900112233,
    message: 'Need functional and aesthetic furniture for a new office setup.',
  },
  {
    id: 8,
    propertyType: 'Retail',
    areaSize: '2200',
    measurement: 'sqft',
    budget: 'High End',
    name: 'Isabella White',
    email: 'isabella.white@example.com',
    contact: 2233445566,
    message: 'Want a premium retail store layout with custom wooden counters.',
  },
]
