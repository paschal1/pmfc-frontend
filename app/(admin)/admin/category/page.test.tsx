import { render, screen, fireEvent } from '@testing-library/react'
import Category from './page'
import { categories } from '../utils/category'
import { useRouter } from 'next/navigation'
import '@testing-library/jest-dom'

jest.mock('next/navigation', () => ({
  useRouter: jest.fn(),
}))

describe('Category Component', () => {
  it('matches the snapshot', () => {
    const { asFragment } = render(<Category />)
    expect(asFragment()).toMatchSnapshot()
  })
  
  test('renders the heading', () => {
    render(<Category />)
    expect(
      screen.getByRole('heading', { level: 1, name: /All Category/i })
    ).toBeInTheDocument()
  })

  test('renders the Category component correctly', () => {
    render(<Category />)
    expect(screen.getByText(/All Category/i)).toBeInTheDocument()
    expect(screen.getByRole('button', { name: /add new/i })).toBeInTheDocument()
    expect(screen.getByTitle('search')).toBeInTheDocument()
  })

  test('navigates to the add category page when clicking the button', () => {
    const push = jest.fn()
    ;(useRouter as jest.Mock).mockReturnValue({push})

    render(<Category />)
    fireEvent.click(screen.getByRole('button', { name: /add new/i }))
    expect(push).toHaveBeenCalledWith('/admin/add-new-category')
  })

  test('renders the correct number of categories', () => {
    render(<Category />)
    const rows = screen.getAllByRole('row')
    expect(rows.length).toBe(categories.length + 1) // Including header row
  })

  test('renders correct category names, images, and icons', () => {
    render(<Category />)
    categories.forEach((category) => {
      expect(screen.getByText(category.name)).toBeInTheDocument()
    })

    // Check images
    const images = screen.getAllByAltText('img')
    expect(images.length).toBe(categories.length)
    images.forEach((img, index) => {
      expect(img).toHaveAttribute('src', categories[index].image)
    })
  })

  test('allows input in the search field', () => {
    render(<Category />)
    const searchInput = screen.getByTitle('search')
    fireEvent.change(searchInput, { target: { value: 'test' } })
    expect(searchInput).toHaveValue('test')
  })
})
