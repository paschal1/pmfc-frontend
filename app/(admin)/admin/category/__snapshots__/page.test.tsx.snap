// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Category Component matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="bg-white flex flex-col pb-[3rem]"
  >
    <div
      class="xl:ml-[20rem] mt-8 bg-[#F2F2F2] flex flex-col px-4 w-[90%] lg:w-[1014px] rounded-xl mx-auto mb-8 pb-8"
    >
      <div
        class="flex items-center justify-between mt-4"
      >
        <h1
          class="font-semibold sm:text-xl text-lg"
        >
          All Category
        </h1>
        <button
          class="px-7 py-2 bg-[#fab702] rounded-[5px] text-white text-[13px] font-semibold hover:text-black hover:opacity-75 active:opacity-60 transition-all duration-500 ease-in-out flex items-center justify-center gap-2"
        >
          <svg
            class="h-[20px] w-[20px]"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 0h24v24H0V0z"
              fill="none"
            />
            <path
              d="M19 3H5a2 2 0 0 0-2 2v14a2 2 0 0 0 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zm0 16H5V5h14v14zm-8-2h2v-4h4v-2h-4V7h-2v4H7v2h4z"
            />
          </svg>
          Add New
        </button>
      </div>
      <div
        class="flex items-center justify-between mt-4"
      >
        <div />
        <div>
          <label
            class="mr-3"
          >
            Search
          </label>
          <input
            class="border bg-inherit border-black focus:outline-none pl-2 h-[35px] w-[150px] rounded-[4px]"
            placeholder=""
            title="search"
            type="text"
          />
        </div>
      </div>
      <div
        class="mt-8 overflow-x-auto relative"
      >
        <table
          class="min-w-full border-collapse"
        >
          <thead>
            <tr
              class="text-left bg-gray-200 rounded-[6px] text-[#4A5568]"
            >
              <th
                class="lg:px-16 px-8 py-2 whitespace-nowrap"
              >
                Category Name
              </th>
              <th
                class="lg:px-16 px-8 py-2 whitespace-nowrap"
              >
                Category Image
              </th>
              <th
                class="lg:px-16 px-8 py-2 whitespace-nowrap"
              >
                Category Icon
              </th>
            </tr>
          </thead>
          <tbody>
            <tr
              class="even:bg-white odd:bg-[#F2F2F2]"
            >
              <td
                class="lg:px-16 px-8 py-3"
              >
                <h1
                  class="text-md text-[#4A5568]"
                >
                  Chair
                </h1>
              </td>
              <td
                class="lg:px-16 px-8 py-3"
              >
                <div
                  class="w-[80px] h-[80px] flex items-center justify-center rounded-xl"
                >
                  <img
                    alt="img"
                    class="h-[60px] w-[60px] object-contain"
                    src="/chair.webp"
                  />
                </div>
              </td>
              <td
                class="lg:px-24 px-16 py-3 ml-4"
              >
                <svg
                  class="text-[#4A5568] text-4xl"
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 24 24"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M8 3C5.79086 3 4 4.79086 4 7V9.12602C2.27477 9.57006 1 11.1362 1 13C1 14.4817 1.8052 15.7734 3 16.4646V19V21H5V20H19V21H21V19V16.4646C22.1948 15.7734 23 14.4817 23 13C23 11.1362 21.7252 9.57006 20 9.12602V7C20 4.79086 18.2091 3 16 3H8ZM18 9.12602C16.2748 9.57006 15 11.1362 15 13H9C9 11.1362 7.72523 9.57006 6 9.12602V7C6 5.89543 6.89543 5 8 5H16C17.1046 5 18 5.89543 18 7V9.12602ZM9 15H15V16H17V13C17 11.8954 17.8954 11 19 11C20.1046 11 21 11.8954 21 13C21 13.8693 20.4449 14.6114 19.6668 14.8865C19.2672 15.0277 19 15.4055 19 15.8293V18H5V15.8293C5 15.4055 4.73284 15.0277 4.33325 14.8865C3.5551 14.6114 3 13.8693 3 13C3 11.8954 3.89543 11 5 11C6.10457 11 7 11.8954 7 13V16H9V15Z"
                  />
                </svg>
              </td>
            </tr>
            <tr
              class="even:bg-white odd:bg-[#F2F2F2]"
            >
              <td
                class="lg:px-16 px-8 py-3"
              >
                <h1
                  class="text-md text-[#4A5568]"
                >
                  Bed
                </h1>
              </td>
              <td
                class="lg:px-16 px-8 py-3"
              >
                <div
                  class="w-[80px] h-[80px] flex items-center justify-center rounded-xl"
                >
                  <img
                    alt="img"
                    class="h-[60px] w-[60px] object-contain"
                    src="/bed.png"
                  />
                </div>
              </td>
              <td
                class="lg:px-24 px-16 py-3 ml-4"
              >
                <svg
                  class="text-[#4A5568] text-4xl"
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 512 512"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M384 240H96V136a40.12 40.12 0 0 1 40-40h240a40.12 40.12 0 0 1 40 40v104zM48 416V304a64.19 64.19 0 0 1 64-64h288a64.19 64.19 0 0 1 64 64v112"
                    fill="none"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="32"
                  />
                  <path
                    d="M48 416v-8a24.07 24.07 0 0 1 24-24h368a24.07 24.07 0 0 1 24 24v8M112 240v-16a32.09 32.09 0 0 1 32-32h80a32.09 32.09 0 0 1 32 32v16m0 0v-16a32.09 32.09 0 0 1 32-32h80a32.09 32.09 0 0 1 32 32v16"
                    fill="none"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="32"
                  />
                </svg>
              </td>
            </tr>
            <tr
              class="even:bg-white odd:bg-[#F2F2F2]"
            >
              <td
                class="lg:px-16 px-8 py-3"
              >
                <h1
                  class="text-md text-[#4A5568]"
                >
                  Cabinet
                </h1>
              </td>
              <td
                class="lg:px-16 px-8 py-3"
              >
                <div
                  class="w-[80px] h-[80px] flex items-center justify-center rounded-xl"
                >
                  <img
                    alt="img"
                    class="h-[60px] w-[60px] object-contain"
                    src="/cabinet.jpg"
                  />
                </div>
              </td>
              <td
                class="lg:px-24 px-16 py-3 ml-4"
              >
                <svg
                  class="text-[#4A5568] text-4xl"
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 24 24"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M21 4c0-1.103-.897-2-2-2H5c-1.103 0-2 .897-2 2v16c0 1.103.897 2 2 2h14c1.103 0 2-.897 2-2V4zM5 4h14v7H5V4zm0 16v-7h14.001v7H5z"
                  />
                  <path
                    d="M14 7h-4V6H8v3h8V6h-2zm0 8v1h-4v-1H8v3h8v-3z"
                  />
                </svg>
              </td>
            </tr>
            <tr
              class="even:bg-white odd:bg-[#F2F2F2]"
            >
              <td
                class="lg:px-16 px-8 py-3"
              >
                <h1
                  class="text-md text-[#4A5568]"
                >
                  Desk
                </h1>
              </td>
              <td
                class="lg:px-16 px-8 py-3"
              >
                <div
                  class="w-[80px] h-[80px] flex items-center justify-center rounded-xl"
                >
                  <img
                    alt="img"
                    class="h-[60px] w-[60px] object-contain"
                    src="/img5.jpg"
                  />
                </div>
              </td>
              <td
                class="lg:px-24 px-16 py-3 ml-4"
              >
                <svg
                  class="text-[#4A5568] text-4xl"
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 256 256"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M244,60H12a12,12,0,0,0,0,24h4V192a12,12,0,0,0,24,0V156H216v36a12,12,0,0,0,24,0V84h4a12,12,0,0,0,0-24ZM40,84h76v48H40Zm176,48H140V84h76ZM100,108a12,12,0,0,1-12,12H68a12,12,0,0,1,0-24H88A12,12,0,0,1,100,108Zm56,0a12,12,0,0,1,12-12h20a12,12,0,0,1,0,24H168A12,12,0,0,1,156,108Z"
                  />
                </svg>
              </td>
            </tr>
            <tr
              class="even:bg-white odd:bg-[#F2F2F2]"
            >
              <td
                class="lg:px-16 px-8 py-3"
              >
                <h1
                  class="text-md text-[#4A5568]"
                >
                  Sofa
                </h1>
              </td>
              <td
                class="lg:px-16 px-8 py-3"
              >
                <div
                  class="w-[80px] h-[80px] flex items-center justify-center rounded-xl"
                >
                  <img
                    alt="img"
                    class="h-[60px] w-[60px] object-contain"
                    src="/img1.jpg"
                  />
                </div>
              </td>
              <td
                class="lg:px-24 px-16 py-3 ml-4"
              >
                <svg
                  class="text-[#4A5568] text-4xl"
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 24 24"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M15 3H9C6.02024 3 3.54782 5.17213 3.07974 8.01937C3.21811 8.00655 3.35829 8 3.5 8C5.98528 8 8 10.0147 8 12.5V14H16V12.5C16 10.0147 18.0147 8 20.5 8C20.6417 8 20.7819 8.00655 20.9203 8.01937C20.4522 5.17213 17.9798 3 15 3ZM21 10.05C20.8384 10.0172 20.6712 10 20.5 10C19.1193 10 18 11.1193 18 12.5V17H16V16H8V17H6V12.5C6 11.1193 4.88071 10 3.5 10C3.32877 10 3.16156 10.0172 3 10.05C1.85888 10.2816 1 11.2905 1 12.5C1 13.0886 1.18173 13.5388 1.39024 14.0554C1.67083 14.7505 1.9999 15.5658 1.99978 17V20C1.99978 20.5523 2.44749 21 2.99978 21H12H21.0002C21.5525 21 22.0002 20.5523 22.0002 20V17C22.0001 15.5658 22.3292 14.7505 22.6098 14.0554C22.8183 13.5388 23 13.0886 23 12.5C23 11.2905 22.1411 10.2816 21 10.05Z"
                  />
                </svg>
              </td>
            </tr>
            <tr
              class="even:bg-white odd:bg-[#F2F2F2]"
            >
              <td
                class="lg:px-16 px-8 py-3"
              >
                <h1
                  class="text-md text-[#4A5568]"
                >
                  Misc
                </h1>
              </td>
              <td
                class="lg:px-16 px-8 py-3"
              >
                <div
                  class="w-[80px] h-[80px] flex items-center justify-center rounded-xl"
                >
                  <img
                    alt="img"
                    class="h-[60px] w-[60px] object-contain"
                    src="/misc.png"
                  />
                </div>
              </td>
              <td
                class="lg:px-24 px-16 py-3 ml-4"
              >
                <svg
                  class="text-[#4A5568] text-4xl"
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 24 24"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M0 0h24v24H0z"
                    fill="none"
                  />
                  <path
                    d="m14.17 13.71 1.4-2.42c.09-.15.05-.34-.08-.45l-1.48-1.16c.03-.22.05-.45.05-.68s-.02-.46-.05-.69l1.48-1.16c.13-.11.17-.3.08-.45l-1.4-2.42c-.09-.15-.27-.21-.43-.15l-1.74.7c-.36-.28-.75-.51-1.18-.69l-.26-1.85a.364.364 0 0 0-.35-.29h-2.8c-.17 0-.32.13-.35.3L6.8 4.15c-.42.18-.82.41-1.18.69l-1.74-.7c-.16-.06-.34 0-.43.15l-1.4 2.42c-.09.15-.05.34.08.45l1.48 1.16c-.03.22-.05.45-.05.68s.02.46.05.69l-1.48 1.16c-.13.11-.17.3-.08.45l1.4 2.42c.09.15.27.21.43.15l1.74-.7c.36.28.75.51 1.18.69l.26 1.85c.03.16.18.29.35.29h2.8c.17 0 .32-.13.35-.3l.26-1.85c.42-.18.82-.41 1.18-.69l1.74.7c.16.06.34 0 .43-.15zM8.81 11c-1.1 0-2-.9-2-2s.9-2 2-2 2 .9 2 2-.9 2-2 2zM21.92 18.67l-.96-.74c.02-.14.04-.29.04-.44 0-.15-.01-.3-.04-.44l.95-.74c.08-.07.11-.19.05-.29l-.9-1.55c-.05-.1-.17-.13-.28-.1l-1.11.45c-.23-.18-.48-.33-.76-.44l-.17-1.18a.216.216 0 0 0-.21-.2h-1.79c-.11 0-.21.08-.22.19l-.17 1.18c-.27.12-.53.26-.76.44l-1.11-.45a.23.23 0 0 0-.28.1l-.9 1.55c-.05.1-.04.22.05.29l.95.74a3.145 3.145 0 0 0 0 .88l-.95.74c-.08.07-.11.19-.05.29l.9 1.55c.05.1.17.13.28.1l1.11-.45c.23.18.48.33.76.44l.17 1.18c.02.11.11.19.22.19h1.79c.11 0 .21-.08.22-.19l.17-1.18c.27-.12.53-.26.75-.44l1.12.45c.1.04.22 0 .28-.1l.9-1.55c.06-.09.03-.21-.05-.28zm-4.29.16a1.35 1.35 0 1 1 .001-2.701 1.35 1.35 0 0 1-.001 2.701z"
                  />
                </svg>
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
